### PR TITLE
Component does no deploy providers in OpenShift 4

### PR DESCRIPTION
**Pull request by ZuGao**
After further tests in OpenShift 4 a bug came up. Crossplane deploys providers with wrong `securityContext`. This pull request address this problem by configuring a `Provider` via `ControllerConfig` object. Also added golden tests with providers for both environments.

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.

### CODE REVIEW

File: component/main.jsonnet [hunk 1]
```diff
@@ -4,14 +4,31 @@ local kube = import 'lib/kube.libjsonnet';
 local inv = kap.inventory();
 
 local params = inv.parameters.crossplane;
+local on_openshift4 = inv.parameters.facts.distribution;
```
Comment (simu) [inline component/main.jsonnet:7]: We generally make `on_openshift4` a boolean, so we can later on use it as just `if on_openshift4 then ...`.

```suggestion
local on_openshift4 = inv.parameters.facts.distribution == 'openshift4';
```

File: component/main.jsonnet [hunk 2]
```diff
@@ -4,14 +4,31 @@ local kube = import 'lib/kube.libjsonnet';
...
+  },
+};
 
 {
   '00_namespace': kube.Namespace(params.namespace),
   [if std.length(params.providers) > 0 then '10_providers']: [
     crossplane.Provider(provider) {
-      spec+: params.providers[provider],
+      spec+: params.providers[provider] +
+             if on_openshift4 == 'openshift4' then controller_config_ref else {},
```
Comment (simu) [inline component/main.jsonnet:28]:
```suggestion
             if on_openshift4 then controller_config_ref else {},
```

File: component/main.jsonnet [hunk 3]
```diff
@@ -4,14 +4,31 @@ local kube = import 'lib/kube.libjsonnet';
...
     }
     for provider in std.objectFields(params.providers)
   ],
   [if params.monitoring.enabled then '20_monitoring']: import 'monitoring.libsonnet',
+  [if on_openshift4 == 'openshift4' then '30_controller_config']: controller_config,
```
Comment (simu) [inline component/main.jsonnet:33]:
```suggestion
  [if on_openshift4 then '30_controller_config']: controller_config,
```

**Review by simu — APPROVED** [review]
LGTM overall, I've not verified that the Crossplane config itself is correct.

**Review by Kidswiss — APPROVED** [review]

**Review by ccremer — APPROVED** [review]

**Pull request merged**